### ISSUE TITLE
[BUG] Duplicate entries when lazyMode:true

### ISSUE THREAD

**Issue by ucefkh** [benwinding/react-admin-firebase#272]
This is a Serious bug I saw for a few years now that wasn't fixed yet

Reproduction:
Use react admin-firebase and set lazyMode:true that's it!

I can't share my Firebase app publicly so you can try it 

Look how bad this is!

![image](https://user-images.githubusercontent.com/1968135/231520471-7a6d02d8-7238-473e-ae96-29a383c2995c.png)

It is affecting us so much and I knew this bug for a few years now that was never fixed

**Comment by ucefkh** [comment]
![image](https://user-images.githubusercontent.com/1968135/231521367-5dd5962c-cc1e-4e5e-a405-fbc6855635ae.png)

**Comment by ucefkh** [comment]
seems people know about it and made this 

https://www.npmjs.com/package/react-admin-firebase-lazy-load/v/3.2.32

**Comment by ucefkh** [comment]
Hello guys,

Any updates on this please?

**Comment by amilosmanli** [comment]
@ucefkh I'm using the `lazy loading` as well, but never encountered this issue before. 

Can you share your `options` that you pass to the `FirebaseDataProvider` ?

`lazyMode:true` is not the correct option btw, you should use 

```
lazyLoading: {
    enabled: true,
  },
```

**Comment by quynguyencanalcircle** [comment]
Hi @ucefkh, I got the same problem.
I make a patch to fix on my side:

ResourceManager.ts:95

```
newDocs.forEach((d) =>{
      const data = parseFireStoreDocument<IResourceItem>(d);
      const existId = resource.list.findIndex(i => i.id === data.id);
      if(existId>=0) {
        resource.list[existId] = data
      } else {
        resource.list.push(data)
      }
    });
```

Hope this help!

**Comment by KonkretneKosteczki** [comment]
I get the same error even without lazy loading enabled. With the newest version of the react admin (4.9.3) and react admin firebase (4.1.2) Every time i refresh the page, entries get appended to the old ones.

**Comment by ucefkh** [comment]
Amazing now fixed in the latest version of react-admin-firebase

https://github.com/benwinding/react-admin-firebase/pull/260

**Comment by ucefkh** [comment]
yes @KonkretneKosteczki  I confirm this is still a bug in 4.1.2 latest version!! so weird!

even though they got getServerCounter now!

**Comment by ucefkh** [comment]
Seems the issue comes from the getQueryCursor which doesnt find the last document id cursor so it fetches the docs again and again

but also we might need to enable indexing for fields??


https://firebase.google.com/docs/firestore/query-data/aggregation-queries

**Comment by ucefkh** [comment]
![image](https://user-images.githubusercontent.com/1968135/233784924-46909ff3-3536-483c-8674-bb4bffae2829.png)

**Comment by ucefkh** [comment]
![image](https://user-images.githubusercontent.com/1968135/233803931-b168d617-f275-43cc-bfe5-8f6f004104e0.png)

**Comment by ucefkh** [comment]
I fixed this issue on my fork 

https://www.npmjs.com/package/react-admin-firebase-01tek

the solution was so dumb!
